common/sampling.cpp

```diff
@@ -195,6 +195,9 @@ static llama_token llama_sampling_sample_impl(
     llama_token id = 0;
     // Get a pointer to the logits
     float * logits = llama_get_logits_ith(ctx_main, idx);
+    if (!logits) {
+        throw std::runtime_error("llama_get_logits_ith failed");
+    }
```

Comment:
This could be a `GGML_ASSERT` instead, since `llama_sampling_prepare` was called first and already does this check. Or maybe it could be omitted.

```suggestion
    GGML_ASSERT(logits); // already checked in llama_sampling_prepare 
```